app/controllers/posts_controller.rb

```diff
@@ -15,7 +15,9 @@ def new
   def create
     @user = User.find(params[:user_id])
     @post = @user.posts.create(post_params)
-    redirect_to user_posts_path(@user)
+    respond_to do |format|
+      format.html { redirect_to user_posts_path(@user), notice: 'Post successfully created!' } if @post
```

Comment:
- [ ] Let's also add a flash message for a not successful path. Kindly add a flash message for when the post is not successfully created.